docs/conf.py

```diff
@@ -104,6 +111,7 @@ def _load_theme(
     'sphinx.ext.autodoc',
     'sphinx.ext.autodoc.typehints',
     'sphinx_rtd_theme',
+    "sphinx.ext.doctest",
```

Comment:
Do we even want doctest? There's already a set of unit tests (being extended from time to time, as necessary), with pytest set up to run them. IIUIC, doctests as snippets of Python code in docstrings, fit pretty much the niche of unit tests of a method or a class - I doubt adding another test runner besides the already established pytest would bring added value.

Reply:
The python snippets are very limited in tmt, so there is not much value for doctest specifically. But `shtest` could be useful to keep the cli output in sync. The purpose of `doctest` and `shtest` are only to check documentation code output to make sure it doesn't fall out-of-sync

Reply:
I'd say the current practice is mostly revolving around calling tmt from various shell scripts under `tests/` directory.

`shtest` sounds interesting, but, I'd still recommend putting both `doctest` and `shtest` aside for now - finishing the build flow changes, plus `linkcheck` which IIUIC does not require anything to be added to docstrings, that would be a fine PR. Since both `doctest` and `shtest` require extra information in docstrings, which we don't have in tmt code yet, adding these two would be IMHO better in standalone PRs, with actual test cases showcasing the usage.